app/controllers/application_controller.rb

```diff
@@ -34,7 +34,7 @@ def require_login
       unless logged_in?
         path_info = request.env['PATH_INFO']
         flash[:warning] = "You must be logged in to access this section"
-        redirect_to '/login?back_to=' + URI.encode(path_info) # halts request cycle
+        redirect_to '/login?back_to=' + URI.encode_www_form(path_info) # halts request cycle
```

Comment:
I'm not sure what the reason for this change is but I think we can use Rails' `to_param` which I think uses `CGI.escape` that's been favoured for (deprecated)`URI.encode`  in Ruby. So instead you could do something like

`redirect_to '/login?back_to=' + path_info.to_param`

What do you think? :)

Reply:
@siaw23 I got it from the docs http://ruby-doc.org/stdlib-2.4.4/libdoc/uri/rdoc/URI/Escape.html. They're both replacements for it, but in this use case I think your solution does makes more sense. Travis likes it too so I made the update!

Reply:
@siaw I saw in one of your [clean up PRs](https://github.com/publiclab/plots2/pull/4902/files#diff-55c5b7aecfb519d0e4880eaf2788eb6eR114) for plots2 that are merged already you wrote: 

```
url = URI.parse(home_url + '?return_to=' + CGI.escape(request.env['PATH_INFO'])).to_s
redirect_to url
```
Do you have any preference for one over the other?

Reply:
### Short Answer:
No.

### Long Answer:
`to_param` and `CGI.escape(request.env['PATH_INFO']))` are essentially the same. If you look closely at how `to_param` is [implemented in Rails](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/to_query.rb#L14), you'll notice it uses `CGI.escape` under the hood; that link is of a file that's required [here](https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/object/to_param.rb). So now if you look [here](https://github.com/rails/rails/blob/a0d7247d1509762283c61182ad82c2eed8d54757/activesupport/lib/active_support/core_ext/object/to_param.rb#L50-L54) you can see how `to_param` uses `to_query` which uses `CGI.escape`. I have no preference, I wrote which one came to mind first but actually think we should use `to_param`, it's nicer. I'll fix it. Thanks!

Reply:
@siaw23 got it, thank you!